VXFM-7463 HC deployments with SuSE-based SVMs are failing

aimers1975 opened the pull request.

gavin-scott commented on lib/puppet/provider/vc_vm/default.rb:
```diff
@@ -1358,11 +1358,11 @@ def deploy_ovf
     end
 
-    if resource[:version]
+    if resource[:version] && resource[:enable_nvdimm] && resource[:enable_nvdimm] == true
```
You don't need the ` && resource[:enable_nvdimm] == true` clause. Also do we need to skip this block of its not enable nvdimm case? I think we're ok in setting the vm version based on input in the manifest.

gavin-scott commented on lib/puppet/provider/vc_vm/default.rb:
```diff
@@ -1358,11 +1358,11 @@ def deploy_ovf
     end
 
-    if resource[:guest_type] && resource[:guestid]
+    if resource[:guest_type] && resource[:guestid] && resource[:enable_nvdimm] && resource[:enable_nvdimm] == true
```
You don't need the && resource[:enable_nvdimm] == true clause.

Also would be good to add a comment that we are skipping reconfiguring the guestid because we think it is only required for the nvdimm case. For other cases the guestid is set when the vm is created.

aimers1975 commented:
@gavin-scott OK, I think I've addressed your feedback...it bothers me we're not doing this the way you ideally want so I'll try to track down some time sometime soon to fix it
